plugins/aws/fix_plugin_aws/resource/base.py

```diff
@@ -220,28 +213,23 @@ def collect_resources(cls: Type[AwsResource], builder: GraphBuilder) -> None:
                 raise
 
     @classmethod
-    def collect(cls: Type[AwsResource], json: List[Json], builder: GraphBuilder) -> List[AwsResource]:
+    def collect(cls: Type[AwsResource], json: List[Json], builder: GraphBuilder) -> None:
         # Default behavior: iterate over json snippets and for each:
         # - bend the json
         # - transform the result into a resource
         # - add the resource to the graph
         # - return a list of resources
         # In case additional work needs to be done, override this method.
-        instances = []
         for js in json:
             if instance := cls.from_api(js, builder):
                 # post process
                 instance.post_process(builder, js)
                 builder.add_node(instance, js)
-                instances.append(instance)
-        return instances
 
     @classmethod
-    def collect_usage_metrics(
-        cls: Type[AwsResource], builder: GraphBuilder, collected_resources: List[AwsResource]
-    ) -> None:
+    def collect_usage_metrics(cls: Type[AwsResource], builder: GraphBuilder) -> Tuple[List, Dict[str, AwsResource], Dict[str, Any]]:  # type: ignore
```

Comment:
This signature is .. complicated and makes me think:
- it is a class method
- returns the list of objects to match
- returns queries and normalizers

Wouldn't it be much simpler, if the method were an instance method, not a class method? This way we would not need the first return value. If you would make the normalization part of the query, you then would only return this:
```
List[AwsCloudwatchQuery]
```

Reply:
```python
def update_resource_metrics(
    resources_map: Dict[str, V@update_resource_metrics],
    cloudwatch_result: Dict[AwsCloudwatchQuery, AwsCloudwatchMetricData],
    metric_normalizers: Dict[str, MetricNormalization]
) -> None
```
`update_resource_metrics` requires  `resources_map`{class.id: class}, `cloudwatch_result`{query: result_data_query} and `metric_normalizers`{metric_name: metric_normalizator}

If we pass `metric_normalizers` in `AwsCloudwatchQuery`, then we can't use `update_resource_metrics`. Because we can't pass normilizer to the function in this case